samza-azure/src/main/java/org/apache/samza/system/eventhub/producer/EventHubSystemProducer.java

```diff
@@ -204,7 +212,15 @@ public synchronized void send(String source, OutgoingMessageEnvelope envelope) {
     checkCallbackThrowable("Received exception on message send");
 
     EventData eventData = createEventData(streamId, envelope);
-    int eventDataLength =  eventData.getBytes() == null ? 0 : eventData.getBytes().length;
+    int eventDataLength = eventData.getBytes() == null ? 0 : eventData.getBytes().length;
+    if (eventDataLength > this.maxMessageSize) {
```

Comment:
Should we not also check the skipBadMessage flag is turned on?

Reply:
We should also disable the handling of badMessage flag in EventProducer

Reply:
skipBadMessage config is datastream config which gets translated into skipLargeMessageConfig for now. 

Added a way to disable the skipLargeMessage. If `skipMessagesLargerThanBytes` is < 0 then we skip the check.